datalad/local/subdatasets.py

```diff
@@ -0,0 +1,438 @@
+        if isinstance(recursion_limit, int) and (recursion_limit <= 0):
+            return
+
+        if set_property:
+            for k, v in set_property:
+                if valid_key.match(k) is None:
+                    raise ValueError(
+                        "key '%s' is invalid (alphanumeric plus '-' only, must "
+                        "start with a letter)" % k)
+        if contains:
+            contains = [rev_resolve_path(c, dataset) for c in assure_list(contains)]
+        for r in _get_submodules(
+                ds, paths, fulfilled, recursive, recursion_limit,
+                contains, bottomup, set_property, delete_property,
+                refds_path):
+            # a boat-load of ancient code consumes this and is ignorant of
+            # Path objects
+            r['path'] = text_type(r['path'])
+            # without the refds_path cannot be rendered/converted relative
+            # in the eval_results decorator
+            r['refds'] = refds_path
+            yield r
+
+
+# internal helper that needs all switches, simply to avoid going through
+# the main command interface with all its decorators again
+def _get_submodules(ds, paths, fulfilled, recursive, recursion_limit,
+                    contains, bottomup, set_property, delete_property,
+                    refds_path):
+    dspath = ds.path
+    if not GitRepo.is_valid_repo(dspath):
+        return
+    modinfo = _parse_gitmodules(ds)
+    # put in giant for-loop to be able to yield results before completion
+    for sm in _parse_git_submodules(ds, paths):
+        if contains and not any(
+                sm['path'] == c or sm['path'] in c.parents for c in contains):
+            # we are not looking for this subds, because it doesn't
+            # match the target path
+            continue
+        # do we just need this to recurse into subdatasets, or is this a
+        # real results?
+        to_report = paths is None \
+            or any(p == sm['path'] or p in sm['path'].parents
+                   for p in paths)
+        sm.update(modinfo.get(sm['path'], {}))
+        if to_report and (set_property or delete_property):
+            # first deletions
+            for dprop in assure_list(delete_property):
+                try:
+                    out, err = ds.repo._git_custom_command(
+                        '', ['git', 'config', '--file', '.gitmodules',
+                             '--unset-all',
+                             'submodule.{}.{}'.format(sm['gitmodule_name'], dprop),
+                            ]
+                    )
+                except CommandError as e:
+                    yield get_status_dict(
+                        'subdataset',
+                        status='impossible',
+                        message=(
+                            "Deleting subdataset property '%s' failed for "
+                            "subdataset '%s', possibly did "
+                            "not exist",
```

Comment:
Given that you bind the exception to `e` above, did you intend to include `e` in the message like you do for the replace exception below?

Reply:
I did not. The exception is rather verbose, but the only meaningful bit is exit code 5, which is a bit pointless and decoded in the actual message. I removed `as e`. Thx.